src/rpm_ostree/cli_status.rs

```diff
@@ -37,83 +38,31 @@ lazy_static::lazy_static! {
     )).unwrap();
 }
 
-/// JSON output from `rpm-ostree status --json`
-#[derive(Clone, Debug, Deserialize)]
-pub struct StatusJSON {
-    deployments: Vec<DeploymentJSON>,
-}
-
-/// Partial deployment object (only fields relevant to zincati).
-#[derive(Clone, Debug, Deserialize)]
-#[serde(rename_all = "kebab-case")]
-pub struct DeploymentJSON {
-    booted: bool,
-    base_checksum: Option<String>,
-    #[serde(rename = "base-commit-meta")]
-    base_metadata: BaseCommitMetaJSON,
-    checksum: String,
-    // NOTE(lucab): missing field means "not staged".
-    #[serde(default)]
-    staged: bool,
-    version: String,
-}
-
-/// Metadata from base commit (only fields relevant to zincati).
-#[derive(Clone, Debug, Deserialize)]
-struct BaseCommitMetaJSON {
-    #[serde(rename = "coreos-assembler.basearch")]
-    basearch: String,
-    #[serde(rename = "fedora-coreos.stream")]
-    stream: String,
-}
-
-impl DeploymentJSON {
-    /// Convert into `Release`.
-    pub fn into_release(self) -> Release {
+impl From<&rpmostree_client::Deployment> for Release {
+    fn from(d: &rpmostree_client::Deployment) -> Self {
         Release {
-            checksum: self.base_revision(),
-            version: self.version,
+            checksum: d
+                .base_checksum
+                .clone()
+                .unwrap_or_else(|| d.checksum.clone()),
+            version: d.version.clone().unwrap_or_default(),
```

Comment:
This looks sketchy, and may break other logic in the app.

I don't know in which cases we can end up with a version-less commit, but it may be good to log a warning and filter it out.